saleor/graphql/app/mutations.py

```diff
@@ -467,35 +442,34 @@ def construct_instance(cls, instance, cleaned_data):
             version=cleaned_data.get("version"),
             token_target_url=cleaned_data.get("tokenTargetUrl"),
             permissions=cleaned_data.get("permissions"),
+            extensions=cleaned_data.get("extensions"),
```

Comment:
Maybe for the fronted it would be better to fallback to an empty list?
```suggestion
            extensions=cleaned_data.get("extensions", []),
```
And then update the GraphQL type to `extensions: [AppManifestExtension!]!`